runtime/appruntime/api/server.go

```diff
@@ -364,3 +373,44 @@ func toUnnamedParams(ps httprouter.Params) UnnamedParams {
 	}
 	return params
 }
+
+// handleTrailingSlashRedirect checks if there's a matching handler
+// with (without) a trailing slash and redirects to it if there is.
+//
+// This is a modified version of the built-in support in httprouter.
+// We can't use the built-in one due to how we handle multiple methods
+// for the same route using Lookup instead of Handle.
+func handleTrailingSlashRedirect(r *httprouter.Router, w http.ResponseWriter, req *http.Request, path string) (handled bool) {
+	// CONNECT does not support redirects.
+	if req.Method == http.MethodConnect {
+		return false
+	}
+
+	// Modify the path to include (exclude) the trailing slash.
+	if len(path) > 1 && path[len(path)-1] == '/' {
+		path = path[:len(path)-1]
+	} else {
+		path += "/"
+	}
+
+	// Check if there is a handler for the modified path.
+	h, _, _ := r.Lookup(req.Method, path)
+	if h == nil {
+		h, _, _ = r.Lookup(wildcardMethod, path)
+	}
+
+	if h == nil {
+		// Couldn't find a handler.
+		return false
+	}
+
+	// Moved Permanently, request with GET method
+	code := http.StatusMovedPermanently
+	if req.Method != http.MethodGet {
+		// Permanent Redirect, request with same method
+		code = http.StatusPermanentRedirect
+	}
```

Comment:
Should this be temporary rather than permanent?

`Permanent` which only impact browser behaviour, rather than say an iOS or Android client - where as it could result in a bad time if the user changes the aPI path and the browser has a permanent redirect stored

Reply:
This was based on the existing trailing slash redirect from httprouter, so I kept this as-is.

Reply:
Yeah I'm still not sure it's the right behaviour for us though.

Will approve; but I'd prefer we changed it to a temporary redirect